rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/SW360RestHealthIndicator.java

```diff
@@ -0,0 +1,107 @@
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.sw360.rest.resourceserver;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import org.apache.thrift.TException;
+import org.eclipse.sw360.datahandler.common.DatabaseSettings;
+import org.eclipse.sw360.datahandler.couchdb.DatabaseInstance;
+import org.eclipse.sw360.datahandler.thrift.ThriftClients;
+import org.eclipse.sw360.datahandler.thrift.health.HealthService;
+import org.eclipse.sw360.datahandler.thrift.health.Status;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class SW360RestHealthIndicator implements HealthIndicator {
+    @Override
+    public Health health() {
+        List<Exception> exceptions = new ArrayList<>();
+        RestState restState = check(exceptions);
+        final String rest_state_detail = "Rest State";
+        if (!restState.isUp()) {
+            Health.Builder builderWithDetails = Health.down()
+                    .withDetail(rest_state_detail, restState);
+            for (Exception exception : exceptions) {
+                builderWithDetails = builderWithDetails.withException(exception);
+            }
+            return builderWithDetails
+                    .build();
+        }
+        return Health.up()
+                .withDetail(rest_state_detail, restState)
+                .build();
+    }
+
+    private RestState check(List<Exception> exception) {
+        RestState restState = new RestState();
+        try {
+            restState.isDbReachable = isDbReachable(exception);
+        } catch (MalformedURLException e) {
+            restState.isDbReachable = false;
+            exception.add(e);
+        }
+        restState.isThriftReachable = isThriftReachable(exception);
+        return restState;
+    }
+
+    private boolean isDbReachable(List<Exception> exception) throws MalformedURLException {
+        DatabaseInstance databaseInstance = makeDatabaseInstance();
+        try {
+            return databaseInstance.checkIfDbExists(DatabaseSettings.COUCH_DB_ATTACHMENTS);
+        } catch (Exception e) {
+            exception.add(e);
+            return false;
+        }
+    }
+
+    private boolean isThriftReachable(List<Exception> exception) {
+        HealthService.Iface healthClient = makeHealthClient();
+        try {
+            final org.eclipse.sw360.datahandler.thrift.health.Health health = healthClient.getHealth();
+            if (health.getStatus().equals(Status.UP)) {
+                return true;
+            } else {
+                exception.add(
+                        new Exception(health.getStatus().toString(),
```

Comment:
The Exception here is a formality that is needed because in this version of the spring boot actuator it does not take throwables in its exception status. The actual exception will unfortunately be wrapped within this exception already. 

I don't think making this a custom exception adds information to the exception, which should be the goal of custom exceptions.